### PR TITLE
Fix: Follower count issues

### LINKED ISSUE #3515: QA Dashboard: Follower count on dashboard doesn't match followers tab data and load in very slowly
# Issues

### 1. The number of followers shown on the dashboard, doesn't match the actual followers shown on the followers tab of the Members page.

Here you see the dashboard data:

image
 
 Here you see the data on the followers tab count:
 
 
image

Another example from a different colony:

image
 
 
 With the actual data being showed on the followers tab of the Members page:
 
 
image

#### Steps to reproduce

1. View the followers and members count on a dashboard.
2. Navigate to the Members page and check the count being displayed on the tabbed menu component.
3. The followers data doesn't match the dashboard data. 

### 2. When visiting a Colony with a large follower base, the followers on the dashboard load slowly and incrementally. With larger colonies with bigger follower bases, this isn't an ideal experience since it takes 18 seconds from skeleton load to load in 1500 members (the final follower total).

Export-1730206303289

#### Steps to reproduce

1. Navigate to a colony with a large follower base. For this test, I use @rdig RC Colony which has 1500 followers.
2. Load the colony and watch after the skeleton loader, the follower count in the top right header load in very slowly. 

## Expected behaviour

* Both the members and followers data displayed on the dashboard should be correct and match across all locations.
* The follower and member data should load in quicker to avoid any wait time or potential confusion on the actual data the colony has to display. 

## Actual behaviour

* The followers data isn't correctly shown on the dashboard.
* Follower count loads in really slowly. 

---

@iamsamgibbs here - Adding an extra issue to this as it overlaps!

The followers page should not have the team switcher.

`The difference between a member and a follower is: a member is someone with reputation or permissions, while a follower is someone who joined the colony.`

Individual teams do not have followers, only the colony as a whole, so we should remove the team switcher here. The same height should be maintained so that the "Members" / "Followers" tabs do not jump up and down when switching tabs.

The follower count should match the follower count on the dashboard and not filter by team even if `?team=3` is manually added to the URL.

image

### PR DESCRIPTION
**Pull request by iamsamgibbs**
## Description

This PR fixes some issues with the members / followers counts and members / followers pages.

The member and follower counts now loads to the correct number without loading in incrementally.

The member and follower counts are the same across the dashboard and members pages.

The team switcher has been removed from the follower page. Followers are no longer filtered by team (as this did not make sense.)

Correctly filter allFollowers to match specifications: https://www.notion.so/colony/Members-e5f7ac2491df4d94b65d9a3ccfefaadf#e3a0e007e6cc4477b03de0d10fe10a81

Added new `useGetColoniesMembersCount` hook.

## Testing

* Step 1 - Spin up the dev environment and run the create-data script
* Step 2 - Navigate to the dashboard - Confirm there are 4 members and 32 followers.

Screenshot 2024-11-11 at 16 34 10

* Step 3 - Run the newly created `create-additional-users` script. It will prompt you to select a number of users to add, enter `500`.

It will then ask for a colony address, copy the planex address from the dashboard and paste it in.

It will ask if you want to verify the users, enter 'N' to decline.

It will then ask if you want to assign user permissions. We do, so enter 'Y'.

It will ask which domain we want to assign permissions in, let's enter '2' to assign permissions in Andromeda.

Finally it will ask which role we want to assign. Let's give the funding role, so enter '5'.

The script will run and add our 500 new users to the colony and give them funding in Andromeda.

* Step 4 - Refresh the dashboard. Check the new 500 users have been added. The member and follower numbers should appear as the correct number as soon as loading is complete (compared to master where they increment by 100 as users are loaded in). **(Note: The members count will take a long time to update whilst the block-ingestor processes all the setUserRole requests. Literally like 15 minutes, so feel free to keep testing once members passes 100 so you can confirm it is not loading in batches of 100.)**

Screenshot 2024-11-11 at 17 29 39

* Step 5 - Click on the follower count to be navigated to the members page with the followers tab pre-selected. Confirm the number of followers matches the number on the dashboard.

Screenshot 2024-11-11 at 17 29 48

* Step 6 - Notice the team switcher does not appear on the followers tab. Change between the two tabs and check there is no shift in content position.

https://github.com/user-attachments/assets/2f81a40a-9913-4acf-8560-91d113da3853

## Diffs

**Changes** 🏗

* The member and follower counts now loads to the correct number without loading in incrementally.

* The member and follower counts are the same across the dashboard and members pages.

* The team switcher has been removed from the follower page. Followers are no longer filtered by team (as this did not make sense.)

* Correctly filter allFollowers to match specifications: https://www.notion.so/colony/Members-e5f7ac2491df4d94b65d9a3ccfefaadf#e3a0e007e6cc4477b03de0d10fe10a81

* Added new `useGetColoniesMembersCount` hook.

Resolves #3515

### CODE REVIEW

**Review by Nortsova — DISMISSED** [review]
Amazing work @iamsamgibbs 👍  

I can confirm that users are not loading in batches of 100:
image

image



The only thing that I have concerns about is the "Followers" page with padding before the tab. It looks not that good to me, but I understand why it was implemented that way :)

**Comment by iamsamgibbs** [comment]
> The only thing that I have concerns about is the "Followers" page with padding before the tab. It looks not that good to me, but I understand why it was implemented that way :)

I agree and discussed it with Mel. We'll keep this for now and open a new issue to change the styling once Mel has been able to put together a new design for this section.

**Review by bassgeta — APPROVED** [review]
Nice solution and thanks for the code comments about what members and followers are in relation to our members and contributors :100: 
My dev env however seems more broken than yours, but it has always been like that :'( 
This is me after a fresh install:
image
However, after running your script, the numbers go up as they should, so this PR does probably make my env more predictable :D Right after:
image
The members page number is then matched  :heavy_check_mark: 
image
No team switcher :heavy_check_mark: 
image

Nice job :sunglasses:

**Comment by iamsamgibbs** [comment]
@Nortsova @bassgeta Apologies, realised an issue with the query filter so have just pushed a small change if you are able to retest at your convenience.

**Review by bassgeta — DISMISSED** [review]
Retested and everything still works as intended! (created 50 additional users and had 0 before due to dev env shennanigans)
image
image

File: scripts/create-additional-users.js [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+          ? 'Error: Please enter only valid role numbers (0,1,2,3,5,6).'
+          : 'Error: Please enter only valid role numbers (2,3,5,6).',
+      );
+      process.exit(1);
+    }
+  }
+}
+
+// Function to validate the inputs
+async function getArguments() {
```
Comment (rdig) [inline scripts/create-additional-users.js:301]: This script is very fancy! Kudos! :1st_place_medal:

**Review by rdig — DISMISSED** [review]
All good. Thanks for fixing this! 


https://github.com/user-attachments/assets/b09f22ae-e994-4401-b187-d5229ad055cf

File: src/hooks/useGetColoniesMembersCount.ts [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    variables: {
+      filter: {
+        or: colonyAddresses.map((colonyAddress) => ({
+          and: [
+            { colonyAddress: { eq: colonyAddress } },
+            {
+              ...filterOptions,
+            },
+          ],
+        })),
```
Comment (mmioana) [inline src/hooks/useGetColoniesMembersCount.ts:33]: Would it make sense to extract this filter into a constant 
```
const colonyAddressesFilter =  {
        or: colonyAddresses.map((colonyAddress) => ({
          and: [
            { colonyAddress: { eq: colonyAddress } },
            {
              ...filterOptions,
            },
          ],
        }))
     }
```
and then use it here `filter: {...colonyAddressesFilter}` as well as below for the `fetchMore`?

**Review by mmioana — DISMISSED** [review]
Really amazing work on this issue @iamsamgibbs 🤩 

The counts seem to update as expected
Screenshot 2024-11-21 at 18 05 58
Screenshot 2024-11-21 at 18 06 57
Screenshot 2024-11-21 at 18 07 26
Screenshot 2024-11-21 at 18 12 50
Screenshot 2024-11-21 at 18 13 05
Screenshot 2024-11-21 at 18 13 09

Left a few refactoring comments, but feel free to decide if it makes sense to handle them now 🙌

**Comment by iamsamgibbs** [comment]
Thanks @mmioana I've included those suggestions!

**Review by rdig — APPROVED** [review]

**Review by bassgeta — APPROVED** [review]
Still looking and working good :muscle: 
image

**Review by mmioana — APPROVED** [review]
Awesome @iamsamgibbs 🌟 thanks for adding those changes!

Everything still works as expected 🙌 

Screenshot 2024-11-22 at 20 16 01
Screenshot 2024-11-22 at 20 16 07

**Pull request merged**